crates/vm/levm/src/opcode_handlers/block.rs

```diff
@@ -206,33 +207,75 @@ impl VM {
     // ADDRESS operation
     pub fn op_address(
         &mut self,
-        _current_call_frame: &mut CallFrame,
+        current_call_frame: &mut CallFrame,
     ) -> Result<OpcodeSuccess, VMError> {
-        unimplemented!();
+        if self.env.consumed_gas + gas_cost::ADDRESS > self.env.gas_limit {
+            return Err(VMError::OutOfGas);
+        }
+
+        let addr = current_call_frame.code_address;
```

Comment:
This works fine in most cases, but not if we are currently in a `DELEGATECALL`, then the address would be msg.sender, not the address of the called contract.

Reply:
Fixed!